EasyDisenchant.lua

```diff
@@ -14,9 +14,43 @@ local _M = {
 	buttonRenderingCache = {},
 };
 
+EasyDisenchantBlacklist = {};
+
 BINDING_HEADER_EASY_DISENCHANT = _M.addonName;
 BINDING_NAME_EASY_DISENCHANT_OPEN = SHOW;
 
+local IsBlacklisted = function(itemLink)
+    for i, BLitemLink in pairs(EasyDisenchantBlacklist) do
+        if(BLitemLink == itemLink) then
+            return true;
+        end
+    end
+    return false;
+end
+
+local Blacklist = function(itemLink)
+    if(EasyDisenchantBlacklist == {}) then
+        nextindex = 0;
```

Comment:
Lua tables are not zero-indexed like in most programming languages, so it's fairly uncommon to use 0 as a table-key when the keys hold no special value in the design.

Reply:
Something else to note here is that you don't declare `nextIndex` in the local scope, so you're actually implicitly setting a global variable here. Not only is that non-optimal, but also could in rare circumstances cause collisions with other addons being bad.